plugins/bibliography/bibparser.py

```diff
@@ -0,0 +1,441 @@
+        
+        md_string = 'title: Publications of ' + author_name.replace('-', ' ') + '\n'
+        md_string += 'template: publications-author\n'
+        md_string += 'author: ' + name +'\n'
+        md_string += 'author_name: ' + author_name.replace('-', ' ') + '\n'
+        md_string += 'groups: ' + ','.join(groups) + '\n'
+        md_string += 'bibkeys: ' + ','.join(bib_keys)
+        md_file_name = './content/pages/publications/' +  name.lower() + '.md'
+        save_md_file(md_file_name, md_string)
+       
+ 
+def create_publications_md(bib_items, author_bib_keys, list_researchers):
+    for bib_key, bib_item in bib_items.items():
+        diag_authors = []
+        groups = set()
+        for name, bib_keys in author_bib_keys.items():
+            for bkey in bib_keys:
+                if bib_key == bkey:
+                    diag_authors.append(name)
+                    for group in list_researchers[name][1]:
+                        groups.add(group)
+                    md_string = 'title: ' + bib_item['title'] + '\n'
+                    md_string += 'template: publication\n'
+                    md_string += 'bibkey: ' + bib_key + '\n'
+                    md_string += 'diag_authors: ' + ','.join(diag_authors) +'\n'
+                    md_string += 'groups: ' + ','.join(groups) + '\n'
+                    if 'journal' in bib_item:
+                        md_string += 'published in: \n' if 'journal' not in bib_item else 'journal: ' + bib_item['journal'] + '\n'
+
+                    if not 'journal' in bib_item and 'booktitle' in bib_item:
+                        md_string += 'booktitle: ' + bib_item['booktitle'] +'\n'
+
+                    md_string += 'pub_type: ' + bib_item['type'] +'\n' 
+                    if 'year' in bib_item:
+                        md_string += 'year: \n' if 'year' not in bib_item else 'year: ' + bib_item['year'] + '\n'
+                    if 'doi' in bib_item:
+                        md_string += 'doi: \n' if 'doi' not in bib_item else 'doi: ' + bib_item['doi'] + '\n'
+                    if 'url' in bib_item:
+                        md_string += 'url: \n' if 'url' not in bib_item else 'url: ' + bib_item['url'] + '\n'
+                    md_string += 'authors: ' + bib_item['authors'] + '\n'
+                    if 'abstract' in bib_item:
+                        md_string += '' if 'abstract' not in bib_item else bib_item['abstract']
+                    md_file_name = './content/pages/publications/' + bib_key + '.md'
+                    save_md_file(md_file_name, md_string)
+
+
+# get bib_items
+@timeit
+def run():
+    bib_items = read_bibtex_file('./content/bib/diag.bib')
+    list_researchers = get_list_researchers('./content/pages/members/')  
+    author_bib_keys = get_publications_by_author(bib_items, list_researchers)
+    save_dict2json('./content/bib/bibitems.json', bib_items)
+    save_dict2json('./content/bib/authorkeys.json', author_bib_keys)
```

Comment:
This function is only 2 lines, does it need a separate function?

Reply:
Maybe a bit overkill, but looks clear in parse_bib_file()